### PR TITLE
Added NXOS provider for ntp_server

### PR DESCRIPTION
**Pull request by jonnytpuppet**

### CODE REVIEW

**Comment by jonnytpuppet** [comment]
Here is the related cisco util gem PR: https://github.com/cisco/cisco-network-node-utils/pull/4
This netdev_stdlib PR is also needed for this to work: https://github.com/puppetlabs/netdev_stdlib/pull/14

File: tests/beaker_tests/ntp_server/ntp_server-provider-defaults.rb [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+# 0 - no changes have occurred, 1 - errors have occurred,
+# 2 - changes have occurred, 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+# 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+# The testcode also uses RegExp pattern matching on stdout or output IO
+# instance attributes of Result object from on() method invocation.
+#
+###############################################################################
+
+# Require UtilityLib.rb and SnmpGroupLib.rb paths.
```
Comment (glennmatthews) [inline tests/beaker_tests/ntp_server/ntp_server-provider-defaults.rb]: s/SnmpGroupLib/ntp_serverlib/

File: tests/beaker_tests/ntp_server/ntp_server-provider-defaults.rb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+testheader = 'ntp_server Resource :: All Attributes Defaults'
+
+# @test_name [TestCase] Executes defaults testcase for ntp_server Resource.
+test_name "TestCase :: #{testheader}" do
+  # @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider test' do
+    # Define PUPPETMASTER_MANIFESTPATH constant using puppet config cmd.
+    UtilityLib.set_manifest_path(master, self)
+
+    # In NX-OS there's no direct configuration of SNMP groups.
```
Comment (glennmatthews) [inline tests/beaker_tests/ntp_server/ntp_server-provider-defaults.rb]: SNMP logic is probably not intentional in ntp_server test cases?

**Comment by jonnytpuppet** [comment]
@glennmatthews thanks for the feedback. I've updated the test as per the comments. Thanks!

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/ntp_server/ntp_server-provider-defaults.rb passed in 32.37 seconds
      Test Suite: tests @ 2015-10-07 01:19:07 +0100

**Pull request merged**